### DISCUSSION TITLE
_data_gas_price': ['Missing data for required field.'], 'l1_da_mode': ['Missing data for required field.']}

### DISCUSSION THREAD

**Discussion by otatataata** [software-mansion/starknet.py#1445] [Q&A]
I'm experiencing an issue when trying to connect to StarkNet using Infura. I'm using the following endpoint: https://starknet-mainnet.infura.io/v3/96f1675247b848d69047593dba2bcb70
Whenever I attempt to make a simple request (like fetching the latest block or estimating gas fees for a transaction), I encounter the following error:

`{'l1_data_gas_price': ['Missing data for required field.'], 'l1_da_mode': ['Missing data for required field.']}`

I’ve double-checked that my Project ID is correct and that my endpoint is active, as shown on my Infura dashboard. However, this error persists. Has anyone else encountered this issue or can advise if any additional configuration is required for StarkNet on Infura?

Thanks in advance for your help!

**Comment by franciszekjob** [comment]
Hey @otatataata , it looks like `l1_data_gas_price` and  `l1_da_mode` fields are missing when  instance of `PendingBlockHeader` is created somewhere in the code. Could you share the code snippet that you're using?
